### PR TITLE
[styled-components] Add type definitions for the babel macro.

### LINKED ISSUE #29792: styled-components@4.0.1: missing macro submodule
- [x] I tried using the `@types/styled-components` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @Igorbek @Igmat @lavoaster

The type definition is missing `styled-components/macro`. All you need to do is re-export/alias/whatever the root `styled-components` type definition under that name, but type definition files make my head spin so I'm not sure the right way to do it myself. In other words, the types should be identical I believe.

From [their docs](https://www.styled-components.com/docs/tooling#babel-macro):
> If your scaffold is set up with babel-plugin-macros, then simply use the new styled-components/macro import instead of styled-components:

### PR DESCRIPTION
**Pull request by gilbsgilbs**
Fixes #29792

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/styled-components/styled-components-tests.tsx [hunk 1]
```diff
@@ -14,6 +14,7 @@ import styled, {
     withTheme,
     ThemeConsumer,
 } from 'styled-components';
+import * as SCMacro from 'styled-components/macro';
```
Comment (Igorbek) [inline types/styled-components/styled-components-tests.tsx]: I would suggest to extract tests of macro to a separate file. Then you would not have to have alias exports like this and use standard `styled` and others.

Comment (Igorbek) [reply types/styled-components/styled-components-tests.tsx]: `styled-components-macro-tests.tsx`?

Comment (gilbsgilbs) [reply types/styled-components/styled-components-tests.tsx]: I wasn't sure if this was possible, but yeah, that's better.

**Review by Igorbek — APPROVED** [review]
Great, thanks! Weird tests were not run

**Review by Igorbek — CHANGES_REQUESTED** [review]
Ah, I might have been wrong about tests, I forgot how to properly organize them here.
According to build it was expecting to have only one name: https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/444749426

I think you should be able to move the test into a new folder `test` to work this around, but not sure.
You can revert to your previous approach if you feel difficulties with that. Sorry for that.

**Comment by gilbsgilbs** [comment]
This is not properly documented, but the `test` directory is indeed [the way to go](https://github.com/Microsoft/types-publisher/issues/318). Thanks for pointing this out. Should be good now.

**Comment by gilbsgilbs** [comment]
> Great, thanks! Weird tests were not run

FYI, tests didn't run because of a huge GitHub outage yesterday : https://status.github.com/messages . They disabled webhooks because they were in a complete inconsistent state due to a network partition :cry: . Everything is back to normal apparently :tada: .

**Review by Igorbek — APPROVED** [review]

**Comment by Igorbek** [comment]
Thank you @gilbsgilbs

**Comment by gilbsgilbs** [comment]
Thanks for the review @Igorbek. 👌

**Pull request merged**